src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/ReferralController.kt

```diff
@@ -31,12 +32,30 @@ class ReferralController(
   private val serviceCategoryService: ServiceCategoryService,
   private val hmppsAuthService: HMPPSAuthService,
 ) {
-  @PostMapping("/draft-referral/{id}/send")
-  fun sendDraftReferral(@PathVariable id: String, authentication: JwtAuthenticationToken): ResponseEntity<SentReferralDTO> {
-    val uuid = parseID(id)
+  @PostMapping("/sent-referral/{id}/assign")
+  fun assignSentReferral(
+    @PathVariable id: UUID,
+    @RequestBody referralAssignment: ReferralAssignmentDTO,
+    authentication: JwtAuthenticationToken,
+  ): SentReferralDTO {
+    val sentReferral = referralService.getSentReferral(id)
+      ?: throw ResponseStatusException(HttpStatus.NOT_FOUND, "sent referral not found [id=$id]")
+
```

Comment:
The retrieval of the existing referral can confined to the service tier. Can the not found exception be raised by the service tier and converted in the controller to a responseStatusException.

Reply:
as in the service method takes an id and throws an error if it doesn't exist?

yeh i'm not against it. if we decide to do it let's do it in another commit though and change all at once.